packages/vite/src/node/server/pluginContainer.ts

```diff
@@ -553,12 +553,26 @@ export async function createPluginContainer(
       }
 
       let combinedMap = this.combinedMap
+      // { mappings: '' }
+      if (
+        combinedMap &&
+        !('version' in combinedMap) &&
+        combinedMap.mappings === ''
+      ) {
+        this.sourcemapChain.length = 0
+        return combinedMap
+      }
+
       for (let m of this.sourcemapChain) {
         if (typeof m === 'string') m = JSON.parse(m)
         if (!('version' in (m as SourceMap))) {
+          // { mappings: '' }
+          if ((m as SourceMap).mappings === '') {
+            combinedMap = { mappings: '' }
+            break
+          }
           // empty, nullified source map
-          combinedMap = this.combinedMap = null
-          this.sourcemapChain.length = 0
+          combinedMap = null
           break
```

Comment:
This part was not correct. `{ mappings: '' }` is different from `null`. So it shouldn't be changed to `null`.
For example, if `this.sourcemapChain` is `[{ mappings: '' }, validSourcemap]`, the combined result should be `{ mappings: '' }` instead of `validSourcemap`.